securedrop_client/export/disk.py

```diff
@@ -0,0 +1,315 @@
+import warnings
+from typing import Callable, NewType, Optional
+
+from PyQt5.QtCore import (
+    QObject,
+    QState,
+    QStateMachine,
+    QTimer,
+    pyqtBoundSignal,
+    pyqtSignal,
+    pyqtSlot,
+)
+
+from .cli import Error as CLIError
+from .service import Service
+
+DEFAULT_POLLING_INTERVAL_IN_MILLISECONDS = 2000
+
+
+class Disk(QObject):
+    """Allows to export files to an ecrypted disk and track its availability."""
+
+    status_changed = pyqtSignal()
+    export_done = pyqtSignal()
+    export_failed = pyqtSignal()
+
+    client_started_watching = pyqtSignal()
+    last_client_stopped_watching = pyqtSignal()
+
+    Status = NewType("Status", str)
+    StatusUnknown = Status("unknown-isw32")
+    StatusLUKSEncrypted = Status("luks-encrypted-8563d")
+    StatusUnreachable = Status("unreachable-ofbu4")
```

Comment:
The suffixes are there to ensure we only compare them as enum constants, not by their string value which is meant to be an opaque string. In other words: if you ever find yourself wondering what the prefix was, you're doing something wrong and should use the constant instead (e.g. `status == Disk.StatusUnreachable`, not `status == "unreachable"`).

I do that as standard practice whenever values are cast automatically instead of raising error when comparing mismatched types.